external_demo_annotator/demo_annotator/vep_annotator.py

```diff
@@ -0,0 +1,398 @@
+            context["feature"].append(row[4])
+            context["feature_type"].append(row[5])
+            context["consequence"].append(row[6])
+            context["cdna_position"].append(row[7])
+            context["cds_position"].append(row[8])
+            context["protein_position"].append(row[9])
+            context["amino_acids"].append(row[10])
+            context["codons"].append(row[11])
+            context["existing_variation"].append(row[12])
+            context["impact"].append(row[13])
+            context["distance"].append(row[14])
+            context["strand"].append(row[15])
+            context["gene"].append(row[17])
+            context["symbol_source"].append(row[18])
+
+        for context in contexts:
+            gene_consequences = []
+            for gene, consequence in zip(
+                context["gene"], context["consequence"], strict=True,
+            ):
+                gene_consequences.append(f"{gene}:{consequence}")
+            context["gene_consequence"] = gene_consequences
+
+            the_consequences: list[str] = []
+            for conseq in context["consequence"]:
+                the_consequences.extend(conseq.split(","))
+
+            context["worst_consequence"] = [sorted(
+                the_consequences, key=lambda x: CONSEQUENCES[x],
+            )[0]]
+            context["highest_impact"] = [sorted(
+                context["impact"], key=lambda x: IMPACTS[x],
+            )[0]]
+
+
+class DemoVEPCacheAnnotator(VEPAnnotatorBase):
+    """Annotation pipeline adapter for dummy_annotate using tempfiles."""
+
+    def __init__(
+        self, pipeline: AnnotationPipeline | None,
+        info: AnnotatorInfo,
+    ):
+        self.vep_cache_dir: Path = info.parameters.get("cache_dir")
+
+        super().__init__(
+            pipeline, info,
+        )
+
+    def _do_batch_annotate(
+        self, annotatables: list[Annotatable | None],
+        contexts: list[dict[str, Any]],
+        batch_work_dir: str | None = None,
+    ) -> list[dict[str, Any]]:
+        if batch_work_dir is None:
+            work_dir = self.work_dir
+        else:
+            work_dir = self.work_dir / batch_work_dir
+        os.makedirs(work_dir, exist_ok=True)
+        with (work_dir / "input.tsv").open("w+t") as input_file, \
+                (work_dir / "output.tsv").open("w+t") as out_file:
+            self.prepare_input(input_file, annotatables)
+            args = [
+                "vep", "-i", input_file.name,
+                "-o", out_file.name,
+                "--tab", "--cache",
+                "--dir", self.vep_cache_dir,
+                "--symbol",
+                "--no_stats",
+                "--force_overwrite",
+            ]
+            subprocess.run(args, check=True)
+            out_file.flush()
+            self.read_output(out_file, contexts)
+
+        self.aggregate_attributes(contexts)
+
+        return contexts
+
+
+class DemoVEPGTFAnnotator(VEPAnnotatorBase):
```

Comment:
We should remove the `Demo` from the name of this annotator. Please rename to `VEPGtfAnnotator` or `VEPFastaGtfAnnotator`?